### PR TITLE
Add MySQL 8, Ubuntu 20.04 and EL8 support

### PR DESCRIPTION
**Pull request by EMH333**
## Description

- Adds support for MySQL 8
- Adds support for Ubuntu 20.04
- Adds support for CentOS 8

Also removes EL6 and Ubuntu 14.04 both of which are EOL/close to EOL

### Issues Resolved

Related to #611 

This also is a prerequisite for #522 (for which I will have another PR very soon)

Supersedes #584 and #626 

### Check List
- [x] All tests pass. See https://github.com/sous-chefs/mysql/blob/master/TESTING.md
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by Xorima** [comment]
Hey @EMH333 

This is awesome!

Can you get something added to the changelog as well please though

Thanks!

**Comment by EMH333** [comment]
Done!

File: libraries/helpers.rb [hunk 1]
```diff
@@ -79,6 +90,7 @@ def default_major_version
       return '5.5' if trusty?
```
Comment (ramereth) [inline libraries/helpers.rb:90]: This should be removed since you removed support for both 5.5 and 14.04. You should also remove the helper.

**Review by ramereth — CHANGES_REQUESTED** [review]
Can you also add support for checking Debian 10 and remove support for Debian 8 while you're at it? Great work thus far!

**Comment by Xorima** [comment]
Can we also have documented the versions of mysql that get removed by this pr in the changelog, just so no one is caught by surprise :)

File: README.md [hunk 2]
```diff
@@ -27,15 +27,15 @@ This cookbook is maintained by the Sous Chefs. The Sous Chefs are a community of
...
-| ubuntu-16.04   |     |     |     | X   |
-| centos-6       | X   | X   | X   | X   |
-| centos-7       |     | X   | X   | X   |
-| fedora         |     |     | X   | X   |
-| openSUSE Leap  |     |     | X   |     |
+|  OS            | 5.6 | 5.7 | 8.0 |
+|----------------|-----|-----|-----|
+| ubuntu-16.04   |     |  X  |     |
+| ubuntu-18.04   |     |  X  |     |
+| ubuntu-20.04   |     |     |  X  |
```
Comment (ramereth) [inline README.md:34]: Please update the list to include Debian support. Also alphabetize it.

File: kitchen.dokken.yml [hunk 3]
```diff
@@ -37,16 +37,23 @@ platforms:
...
 
-  - name: centos-6
+  - name: debian-10
     driver:
-      image: dokken/centos-6
-      pid_one_command: /sbin/init
+      image: dokken/debian-10
+      pid_one_command: /bin/systemd
+      intermediate_instructions:
+        - RUN /usr/bin/apt-get update
```
Comment (ramereth) [inline kitchen.dokken.yml:45]: Please remove debian-8 above here.

**Review by ramereth — CHANGES_REQUESTED** [review]

**Review by Xorima — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [9.0.0](https://github.com/sous-chefs/mysql/releases/tag/9.0.0)

**Pull request merged**